src/rashdf/plan.py

```diff
@@ -1117,6 +1121,106 @@ def reference_lines_timeseries_output(self) -> xr.Dataset:
         """
         return self.reference_timeseries_output(reftype="lines")
 
+    def observed_timeseries_input(self, vartype: str = "Flow") -> xr.Dataset:
+        """Return observed timeseries input data for reference lines and points from a HEC-RAS HDF plan file.
+
+        Parameters
+        ----------
+        vartype : str, optional
+            The type of observed data to retrieve. Must be either "Flow" or "Stage".
+            (default: "Flow")
+
+        Returns
+        -------
+        Dict[Site: str, xr.DataArray]
+            An xarray Dataset with observed timeseries data for both reference lines and reference points.
+
+            Coordinates:
+            - 'time': datetime64[ns]
+
+            Data variables:
+            - 'Flow' or 'Stage': float64
+        """
+
+        # Decode the contents of the DataFrame from utf-8
+        def decode_bytes(val):
+            if isinstance(val, bytes):
+                return val.decode("utf-8")
+            return val
+
+        if vartype == "Flow":
+            output_path = self.OBS_FLOW_OUTPUT_PATH
+        elif vartype == "Stage":
+            output_path = self.OBS_STAGE_OUTPUT_PATH
+        else:
+            raise ValueError('vartype must be either "Flow" or "Stage".')
+
+        observed_group = self.get(output_path)
+        if observed_group is None:
+            raise RasPlanHdfError(
+                f"Could not find HDF group at path '{output_path}'."
+                f" Does the Plan HDF file contain reference {vartype[:-1]} output data?"
+            )
+        print(f"observed_group: {observed_group}")
+        if "Attributes" in observed_group.keys():
+            attr_path = observed_group["Attributes"]
+            print(f"attr_path: {attr_path}")
+            attrs_df = pd.DataFrame(attr_path[:])
+            # Apply the decoding function to each element in the DataFrame
+            attrs_df = attrs_df.map(decode_bytes)
+            if vartype == "Flow":
+                attrs_df["Units"] = "cfs"
+            elif vartype == "Stage":
+                attrs_df["Units"] = "ft"
+            else:
+                attrs_df["Units"] = "Unknown"
+        das = {}
```

Comment:
Rather than an xarray DataArray, this is returning a dictionary of xarray DataArrays.
```python
>>> result = plan_hdf.observed_timeseries_input()
observed_group: <HDF5 group "/Event Conditions/Observed Data/Flow" (2 members)>
attr_path: <HDF5 dataset "Attributes": shape (1,), type "|V68">
>>> result
{'Denton-Justin_RL': <xarray.DataArray 'Flow' (time: 1931)> Size: 15kB
array([ 1.96000004,  1.96000004,  1.79999995, ..., 10.5       ,
       10.5       , 10.5       ])
Coordinates:
  * time     (time) datetime64[ns] 15kB 2015-10-22T23:45:00 ... 2015-11-12T00...
Attributes:
    units:     cfs
    hdf_path:  Event Conditions/Observed Data/Flow/Ref Line: Denton-Justin_RL}
>>> type(result)
<class 'dict'>
```
Instead of returning a dictionary of DataArray objects with the refline/refpoint name as the key, it would make more sense in my opinion to have the reference line/point name be an additional dimension of the returned DataArray. This is how `reference_lines_timeseries_output` works:
```python
>>> phdf = RasPlanHdf("/mnt/c/temp/ElkMiddle.p01.hdf")
>>> phdf.reference_lines_timeseries_output()
<xarray.Dataset> Size: 28kB
Dimensions:        (time: 577, refln_id: 5)
Coordinates:
  * time           (time) datetime64[ns] 5kB 1996-01-14T12:00:00 ... 1996-02-...
  * refln_id       (refln_id) int64 40B 0 1 2 3 4
    refln_name     (refln_id) <U17 340B 'Herold Gage' ... 'DS Sutton Gage'
    mesh_name      (refln_id) <U9 180B 'ElkMiddle' 'ElkMiddle' ... 'ElkMiddle'
Data variables:
    Flow           (time, refln_id) float32 12kB dask.array<chunksize=(577, 5), meta=np.ndarray>
    Water Surface  (time, refln_id) float32 12kB dask.array<chunksize=(577, 5), meta=np.ndarray>
>>> phdf.reference_lines_timeseries_output()["Flow"]
<xarray.DataArray 'Flow' (time: 577, refln_id: 5)> Size: 12kB
dask.array<array, shape=(577, 5), dtype=float32, chunksize=(577, 5), chunktype=numpy.ndarray>
Coordinates:
  * time        (time) datetime64[ns] 5kB 1996-01-14T12:00:00 ... 1996-02-07T...
  * refln_id    (refln_id) int64 40B 0 1 2 3 4
    refln_name  (refln_id) <U17 340B 'Herold Gage' ... 'DS Sutton Gage'
    mesh_name   (refln_id) <U9 180B 'ElkMiddle' 'ElkMiddle' ... 'ElkMiddle'
Attributes:
    units:     cfs
    hdf_path:  Results/Unsteady/Output/Output Blocks/Base Output/Unsteady Tim...
```
Do reference lines allow observed stage data? If so, it might make sense to return `xr.Dataset` objects, which would allow a single returned object with both flow and stage DataArrays, like `reference_lines_timeseries_output`. Also, if reference lines allow observed stage data, I'd guess that it's stored in the `Stage` group... which might mean needing to change the whole approach here from Flow/Stage to Reflines/Refpoints...

Reply:
Yep, both stage and flow may be captured by ref lines for observed data. This function has now been updated to return an xarray as suggested.